pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -1,19 +1,107 @@
 package buildplan
 
 import (
+	"strings"
+
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/locale"
 	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/model"
+	platformModel "github.com/ActiveState/cli/pkg/platform/model"
 	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
 	"github.com/go-openapi/strfmt"
 )
 
+type ArtifactListing struct {
+	build            *model.Build
+	runtimeClosure   artifact.Map
+	buildtimeClosure artifact.Map
+	artifactIDs      []artifact.ArtifactID
+}
+
+func NewArtifactListing(build *model.Build, buildtimeClosure bool) (*ArtifactListing, error) {
+	al := &ArtifactListing{build: build}
+	if buildtimeClosure {
+		buildtimeClosure, err := newMapFromBuildPlan(al.build, true)
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not create buildtime closure")
+		}
+		al.buildtimeClosure = buildtimeClosure
+	} else {
+		runtimeClosure, err := newMapFromBuildPlan(al.build, false)
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not create runtime closure")
+		}
+		al.runtimeClosure = runtimeClosure
+	}
+
+	return al, nil
+}
+
+func (al *ArtifactListing) RuntimeClosure() (artifact.Map, error) {
+	if al.runtimeClosure != nil {
+		return al.runtimeClosure, nil
+	}
+
+	runtimeClosure, err := newMapFromBuildPlan(al.build, false)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not create runtime closure")
+	}
+	al.runtimeClosure = runtimeClosure
+
+	return runtimeClosure, nil
+}
+
+func (al *ArtifactListing) BuildtimeClosure() (artifact.Map, error) {
+	if al.buildtimeClosure != nil {
+		return al.buildtimeClosure, nil
+	}
+
+	buildtimeClosure, err := newMapFromBuildPlan(al.build, true)
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not create buildtime closure")
+	}
+	al.buildtimeClosure = buildtimeClosure
+
+	return buildtimeClosure, nil
+}
+
+func (al *ArtifactListing) ArtifactIDs(buildtimeClosure bool) ([]artifact.ArtifactID, error) {
+	if al.artifactIDs != nil {
+		return al.artifactIDs, nil
+	}
+
+	var artifactMap artifact.Map
+	var err error
+	if buildtimeClosure {
+		artifactMap, err = al.BuildtimeClosure()
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not calculate buildtime closure")
+		}
+	} else {
+		artifactMap, err = al.RuntimeClosure()
+		if err != nil {
+			return nil, errs.Wrap(err, "Could not calculate runtime closure")
+		}
+	}
+
+	for _, artifact := range artifactMap {
+		al.artifactIDs = append(al.artifactIDs, artifact.ArtifactID)
+	}
+
+	return al.artifactIDs, nil
+}
+
 // NewMapFromBuildPlan creates an artifact map from a build plan. It creates a
 // lookup table and calls the recursive function buildMap to build up the
 // artifact map by traversing the build plan from the terminal targets through
 // all of the runtime dependencies for each of the artifacts in the DAG.
-func NewMapFromBuildPlan(build *model.Build) (artifact.Map, error) {
+
```

Comment:
This newline breaks up the documentation comment. Was that intentional? If not, please add a `//` prefix to this empty line. Also, the name has changed to `newMapFromBuildPlan`, so please update the comment on line 95.

Reply:
Yeah, that was intentional as the two blocks are discussing different concepts related to the function.